packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ModelConfigurationController.java

```diff
@@ -195,10 +206,12 @@ public ResponseEntity<Model> getConfiguredModel(
 			if (model.isEmpty()) {
 				throw new ResponseStatusException(HttpStatus.NOT_FOUND, messages.get("model.not-found"));
 			}
-
-			// TODO: Apply configuration to the Model here. Should this use ModelConfigurationLegacy??
-
-			throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED);
+			updateModelParameters(
+					model.get().getParameters(), modelConfiguration.get().getParameterSemanticList());
+			updateModelInitials(
+					model.get().getInitials(), modelConfiguration.get().getInitialSemanticList());
+			updateModelObservables(model.get().getObservables(), modelConfiguration.get().getObservableSemanticList());
```

Comment:
Why are we using `update` instead of `set` here?
A `model-configuration` should match the model that is tied to the `modelId` so there is no reasons to do some matching of information here? Or I am missing something?

Reply:
Ill rename the function to set instead of update.

This jist of these functions are to set the semantics in an AMR structure from the model it is derived from.  The model semantics would have additional properties that we do not get from a model configuration (i.e. name, description, concept...).  I am just simply setting the certain properties that we have in the model configuration to the model (value, distribution, expression...).

Reply:
@YohannParis fyi it isn't super clear that the config will match the model exactly, you may have extra observables and I don't think we have made it clear where observables should be defined. May be worth to clear up after the 1st around.

Reply:
@blanchco in that case you are right, what ever name you decide is good with me. No worries.
@mwdchang the Observable are set on the Model operator, and are port of the model object, no?